File: client/src/components/users/UserProfile.js
import React, { useEffect, useState } from 'react';
import { exitFromProfile, getUser, updateUser } from '../../services/usersService';
import ModalButton from '../ModalButton';
import UserProfileCreation from './UserProfileCreation';
import UserView from './UserView';

const UserProfile = () => {
  const [user, setUser] = useState({
    id: 0,
    name: '',
    email: '',
    password: '',
    description: '',
    photo: '',
  });

  useEffect(() => {
    const fetchUser = async () => {
      const data = await getUser();
      setUser(data);
    };

    fetchUser();
  }, []);

  const updateUserView = (newUser) => {
    setUser(newUser);
    updateUser(newUser);
  }

  return (
    <div>
      <div style={{
            display: 'flex',
            justifyContent: 'flex-end'}}>
            <ModalButton
              modalContent = {<UserProfileCreation
                            user={user}
                            setAction={updateUserView} />}
              title = "Edit profile"
              btnName="Edit"/>
            <button 
              className='btn btn-secondary'
              onClick={() => exitFromProfile()}>
                Exit
            </button>
          </div>
          <UserView user={user} isProfile={true}/> 
    </div>
  );
};

export default UserProfile;

File: Api/Models/UserProfile.cs
namespace Api.Models;

/// <summary>
/// Представляет профиль пользователя, содержащий основную информацию о пользователе.
/// </summary>
public class UserProfile
{
    /// <summary>
    /// Получает или задает уникальный идентификатор профиля пользователя.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Получает или задает имя пользователя.
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// Получает или задает адрес электронной почты пользователя.
    /// </summary>
    public string Email { get; set; }

    /// <summary>
    /// Получает или задает описание пользователя. Это может быть краткая биография или другая информация о пользователе.
    /// </summary>
    public string Description { get; set; }

    /// <summary>
    /// Получает или задает фотографию пользователя. Может хранить путь к изображению или объект изображения.
    /// </summary>
    public object Photo { get; set; }

    /// <summary>
    /// Получает или задает количество подписчиков пользователя.
    /// </summary>
    public int SubsCount { get; set; }
}